### PR TITLE
Linting workflow & Dependabot integration

### PR DESCRIPTION
**Pull request by ToasterUwU**
As promised in #13

### CODE REVIEW

**Comment by ToasterUwU** [comment]
@rsnitsch Have a look, and feel free to merge if you are ok with this. 

I use this exact same thing in many of my Python projects, it helps a lot to minimize upkeep and also prevents many of the basic issues that can arise ( unbound vars, wrong types, etc. )

**Comment by ToasterUwU** [comment]
In case you arent familiar, everything used in here is free to use, so even tho this takes processing power, Github doesnt charge you for it.

**Comment by ToasterUwU** [comment]
Just realized i need to change it to use pipenv instead ( I don't use that )

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: Lint
+
+on:
+  push:
+    branches:
+      - master
```
Comment (rsnitsch) [inline .github/workflows/lint.yml:6]: I guess this has to be changed to `develop` in case of `py3createtorrent`.

**Comment by rsnitsch** [comment]
Very promising, thank you.

File: .github/workflows/lint.yml (hunk 1 above)
Comment (ToasterUwU) [reply .github/workflows/lint.yml]: Good oberservation, forgot about that as well. Fixed it now.

**Comment by ToasterUwU** [comment]
Ok, i have never used pipenv, i need to test it really quick on my fork, just to make sure it runs fine. After that you can merge it.

**Comment by rsnitsch** [comment]
> Ok, i have never used pipenv, i need to test it really quick on my fork, just to make sure it runs fine. After that you can merge it.

Thank you for the extra diligence. 👌

**Comment by ToasterUwU** [comment]
@rsnitsch Take another look and merge if you want to. I tested this on my fork, and it worked.
Meaning it screamed at me for the fact. I don't have the fixes from the #16 PR, but with them its smooth sailing.

**Comment by rsnitsch** [comment]
Thank you very much! I merge this first because I want to see it screaming. ;-)

**Comment by ToasterUwU** [comment]
@rsnitsch Btw, it might make sense to add all Python versions you want to officially support to the lint workflow. Just add them to the OS matrix thing next to 3.10 and 3.11. I recommend putting 3.8 in at least because that's pre-built-in type hinting.

**Pull request merged**